test/test_retriever.py

```diff
@@ -243,6 +345,43 @@ def test_dataset_names_upstream():
     keyword_datasets = rt.get_dataset_names_upstream(keywords=['plants'])
     assert 'biodiversity-response' in keyword_datasets
 
+def test_socrata_autocomplete_search():
+    """Check if autocomplete search returns a list of names or not"""
+    names = rt.socrata_autocomplete_search(["building", "permits"])
+    assert isinstance(names, list) and (len(names) != 0)
+    names = rt.socrata_autocomplete_search([" "])
+    assert isinstance(names, list) and (len(names) == 0)
+    names = rt.socrata_autocomplete_search(["fshing"])
+    assert isinstance(names, list) and (len(names) == 0)
+
+def test_socrata_dataset_info():
+    """Check if socrata dataset info returns metadata for a dataset name"""
+    resource = rt.socrata_dataset_info("Building Permits")
+    assert isinstance(resource, list) and (len(resource) != 0) and (resource[0]["id"] != None)
```

Comment:
`all ([isinstance(resource, list), len(resource), resource[0]["id"]])`
we can also simplify these as above